pkg/vault/contracts/WrappedBalancerPoolTokenFactory.sol

```diff
@@ -10,31 +10,39 @@ import {
     IWrappedBalancerPoolTokenFactory
 } from "@balancer-labs/v3-interfaces/contracts/vault/IWrappedBalancerPoolTokenFactory.sol";
 
+import { SingletonAuthentication } from "./SingletonAuthentication.sol";
 import { WrappedBalancerPoolToken } from "./WrappedBalancerPoolToken.sol";
 
 /// @notice Factory contract for creating wrapped Balancer pool tokens
-contract WrappedBalancerPoolTokenFactory is IWrappedBalancerPoolTokenFactory {
-    IVault internal immutable _vault;
+contract WrappedBalancerPoolTokenFactory is IWrappedBalancerPoolTokenFactory, SingletonAuthentication {
+    bool private _isDisabled;
     mapping(address => address) internal _wrappedTokens;
 
-    constructor(IVault vault) {
-        _vault = vault;
+    constructor(IVault vault) SingletonAuthentication(vault) {
+        // solhint-disable-previous-line no-empty-blocks
+    }
+
+    modifier notDisabled() {
+        if (_isDisabled) {
+            revert Disabled();
+        }
+        _;
     }
 
     /// @inheritdoc IWrappedBalancerPoolTokenFactory
-    function createWrappedToken(address balancerPoolToken) external returns (address) {
+    function createWrappedToken(address balancerPoolToken) external notDisabled authenticate returns (address) {
```

Comment:
I think I'd keep wrapper creation permissionless; it's pretty harmless really. Do you see any reason to make it permissioned?